pkg/master/tunneler/ssh.go

```diff
@@ -162,7 +162,8 @@ func (c *SSHTunneler) SecondsSinceSSHKeySync() int64 {
 func (c *SSHTunneler) installSSHKeySyncLoop(user, publicKeyfile string) {
 	go wait.Until(func() {
 		if c.InstallSSHKey == nil {
-			glog.Error("Won't attempt to install ssh key: InstallSSHKey function is nil")
+			atomic.StoreInt64(&c.lastSSHKeySync, c.clock.Now().Unix())
+			glog.Info("Won't attempt to install ssh key: InstallSSHKey function is nil")
```

Comment:
s/Info/Infof ?

Reply:
@xiangpengzhao What for? Theres no formatting happening

Reply:
Ah, yes :)

Reply:
seems like it would be better to skip checking lastSSHKeySync if InstallSSHKey is nil, rather than put bogus data in the variable

Reply:
The Tunneler interface does not know the `InstallSSHKey` field obviously. IMO the Tunneler logic leaks too much about sync already.

Reply:
so we run a noop goroutine forever that logs and updates this timestamp instead?

Reply:
How about moving the `SecondsSince...` funcs into the implementation and put a `Healthy() bool` or `Healthy() error` into the interface?

Reply:
Can this change go now?